styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java

```diff
@@ -208,7 +168,120 @@ private void ensureSecrets(WorkflowInstance workflowInstance, RunSpec runSpec) {
         LOG.info("[AUDIT] Workflow {} refers to secret {}",
                  workflowInstance.workflowId(), specSecret.name());
       }
+    });
+  }
+
+  private void ensureServiceAccountKeySecret(WorkflowInstance workflowInstance, RunSpec runSpec) {
```

Comment:
So as we discussed, this method is supposed to be `synchronized` as the first simplest approach.